executor/main.go

```diff
@@ -231,25 +222,40 @@ func main() {
 	}
 	defer closer.Close()
 
-	if *transport == "rest" {
-		clientRest, err := rest.NewJSONRestClient(*protocol, *sdepName, predictor, annotations)
-		if err != nil {
-			log.Fatalf("Failed to create http client: %v", err)
-		}
-		logger.Info("Running http server ", "port", *httpPort)
-		runHttpServer(logger, predictor, clientRest, *httpPort, false, serverUrl, *namespace, *protocol, *sdepName, *prometheusPath)
-	} else {
-		logger.Info("Running http probes only server ", "port", *httpPort)
-		go runHttpServer(logger, predictor, nil, *httpPort, true, serverUrl, *namespace, *protocol, *sdepName, *prometheusPath)
-		logger.Info("Running grpc server ", "port", *grpcPort)
-		var clientGrpc seldonclient.SeldonApiClient
-		if *protocol == "seldon" {
-			clientGrpc = seldon.NewSeldonGrpcClient(predictor, *sdepName, annotations)
-		} else {
-			clientGrpc = tensorflow.NewTensorflowGrpcClient(predictor, *sdepName, annotations)
-		}
-		runGrpcServer(logger, predictor, clientGrpc, *grpcPort, serverUrl, *namespace, *protocol, *sdepName, annotations)
+	// Create a listener at the desired port.
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", *port))
+	if err != nil {
+		log.Fatalf("failed to create listener: %v", err)
+	}
+	defer lis.Close()
+
+	// Create a cmux object.
+	tcpm := cmux.New(lis)
 
+	// Declare the match for different services required.
+	httpl := tcpm.Match(cmux.HTTP1Fast())
+	grpcl := tcpm.MatchWithWriters(
+		cmux.HTTP2MatchHeaderFieldSendSettings("content-type", "application/grpc"))
+
+	clientRest, err := rest.NewJSONRestClient(*protocol, *sdepName, predictor, annotations)
+	if err != nil {
+		log.Fatalf("Failed to create http client: %v", err)
 	}
+	logger.Info("Running HTTP server ", "port", *port)
+	go runHttpServer(httpl, logger, predictor, clientRest, *port, false, serverUrl, *namespace, *protocol, *sdepName, *prometheusPath)
```

Comment:
`probesOnly` is no longer need? Can we remove it and also from the restapi:
```
if !probesOnly {
		serverMetrics = metric.NewServerMetrics(predictor, deploymentName)
	}
```

Reply:
Fixed

Reply:
https://github.com/SeldonIO/seldon-core/issues/1856